pkg/allocator/allocator_test.go

```diff
@@ -517,6 +524,61 @@ func TestObserveAllocatorChanges(t *testing.T) {
 //	wg.Wait()
 //}
 
+func TestHandleK8sDelete(t *testing.T) {
+	masterKeyRecreateMaxInterval = time.Millisecond
+	backend := newDummyBackend()
+	alloc := &Allocator{
+		localKeys: &localKeys{
+			keys: map[string]*localKey{},
+			ids:  map[idpool.ID]*localKey{},
+		},
+		enableMasterKeyProtection: true,
+		idPool:                    idpool.NewIDPool(1, 100),
+		backend:                   backend,
+	}
+
+	lc := &localKey{
+		val: 1234,
+		key: TestAllocatorKey("foo"),
+	}
+	alloc.localKeys.keys["foo"] = lc
+	alloc.localKeys.ids[1234] = lc
+
+	var counter atomic.Uint32
+	backend.updateKey = func(ctx context.Context, id idpool.ID, key AllocatorKey) error {
+		defer counter.Add(1)
+		if counter.Load() <= 1 {
+			return fmt.Errorf("updateKey failed")
+		}
+		return nil
+	}
+
+	// 1. Simulate a delete event, where master key protection is enabled
+	// and the identity is owned locally.
+	assert.Nil(t, backend.identities[1234])
+	cache := newCache(alloc)
+	cache.OnDelete(1234, TestAllocatorKey("foo"))
+	assert.Eventually(t, func() bool {
+		return backend.identities[1234] != nil
+	}, time.Second, time.Millisecond*100)
+	assert.Eventually(t, func() bool {
+		return counter.Load() == 3
+	}, time.Millisecond*100, time.Millisecond)
+
+	// 2. Simulate a delete event, where master key protection is enabled
+	// and the identity is owned locally.
+	alloc.enableMasterKeyProtection = false
+	cache.OnDelete(1234, TestAllocatorKey("foo"))
+	assert.NotContains(t, backend.identities, 1234, "no master key protection should delete key")
```

Comment:
s/1234/idpool.ID(1234)/, otherwise the check is always successful due to type mismatch.